tests/Mutator/ZeroIteration/Foreach_Test.php

```diff
@@ -24,28 +24,44 @@ public function test_mutator($input, $expected = null)
         $this->doTest($input, $expected);
     }
 
-    public function provideMutationCases(): array
+    public function provideMutationCases(): \Generator
     {
-        return [
-            'It mutates to new array in foreach' => [
-                <<<'PHP'
+        yield 'It mutates to new array in foreach' => [
+            <<<'PHP'
 <?php
 
 $array = [1, 2];
-
 foreach ($array as $value) {
 }
 PHP
-                ,
-                <<<'PHP'
+            ,
+            <<<'PHP'
 <?php
 
 $array = [1, 2];
 foreach (array() as $value) {
 }
 PHP
-                ,
-            ],
+            ,
+        ];
+        yield 'It does not change whether items were passed by reference' => [
```

Comment:
Once again, to make sure the diffs stay relevant of what we are doing, and not to cause confusion.